datetimepicker.js

```diff
@@ -132,12 +147,32 @@ angular.module('ui.bootstrap.datetimepicker',
           }
         ],
         link: function(scope, element) {
+          var firstTimeAssign = true;
+
           scope.$watch(function() {
             return scope.ngModel;
-          }, function(ngModel) {
+          }, function(newTime) {
             // if a time element is focused, updating its model will cause hours/minutes to be formatted by padding with leading zeros
             if (!element.children()[1].contains(document.activeElement)) {
-              scope.time = new Date(ngModel);
+
+              if (newTime == null || newTime === '') { // if the newTime is not defined
+                if (firstTimeAssign) { // if it's the first time we assign the time value
+                  // create a new default time where the hours, minutes, seconds and milliseconds are set to 0.
+                  newTime = new Date;
```

Comment:
`new Date` is the functionally the same as `new Date()` but I'm happy to change this.